sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientCertificateCredential.java

```diff
@@ -148,8 +148,10 @@ public Mono<AccessToken> getToken(TokenRequestContext request) {
     public AccessToken getTokenSync(TokenRequestContext request) {
         try {
             AccessToken token = identitySyncClient.authenticateWithConfidentialClientCache(request);
-            LoggingUtil.logTokenSuccess(LOGGER, request);
-            return token;
+            if (token != null) {
+                LoggingUtil.logTokenSuccess(LOGGER, request);
+                return token;
```

Comment:
Can msal return null here ? I don't think they do, but if they do, its a bug.
If we got a token response, it is expected to be a non null and a success scenario.
So, the null check seems redundant.

Reply:
I added returning null in the case where we do not get a token from the cache.

Reply:
if there's no token to return, then an exception will get thrown I believe from Msal.
So, in what scenario do we get null back for token ?